packages/react-components/src/templates/__tests__/WelcomePage.test.tsx

```diff
@@ -1,28 +1,33 @@
-import { render } from '@testing-library/react';
+import { screen, render } from '@testing-library/react';
 
 import WelcomePage from '../WelcomePage';
 import { noop } from '../../utils';
 
 it('renders the signin page', () => {
   const handleClick = jest.fn();
-  const { getByRole } = render(<WelcomePage onClick={handleClick} />);
-  expect(getByRole('button').textContent).toMatchInlineSnapshot(`"Sign in"`);
+  render(<WelcomePage onClick={handleClick} />);
+  expect(screen.getByRole('button').textContent).toMatchInlineSnapshot(
+    `"Sign in"`,
+  );
 });
 
 it('renders the signup page', () => {
   const handleClick = jest.fn();
-  const { getByRole } = render(
-    <WelcomePage allowSignup onClick={handleClick} />,
-  );
-  expect(getByRole('button').textContent).toMatchInlineSnapshot(
+  render(<WelcomePage allowSignup onClick={handleClick} />);
+  expect(screen.getByRole('button').textContent).toMatchInlineSnapshot(
     `"Activate account"`,
   );
 });
 
 it('shows an auth failed error message', () => {
-  const { queryByText, rerender } = render(<WelcomePage onClick={noop} />);
-  expect(queryByText(/problem/i)).not.toBeInTheDocument();
+  render(<WelcomePage onClick={noop} />);
+  expect(screen.queryByText(/problem/i)).not.toBeInTheDocument();
+
+  render(<WelcomePage authFailed={'invalid'} onClick={noop} />);
+  expect(screen.queryByText(/problem/i)).toBeVisible();
+});
 
-  rerender(<WelcomePage authFailed onClick={noop} />);
-  expect(queryByText(/problem/i)).toBeVisible();
+it('shows an alumni no-access error message', () => {
+  render(<WelcomePage authFailed={'alumni'} onClick={noop} />);
+  expect(screen.queryByText(/alumni/i)).toBeVisible();
```

Comment:
```suggestion
 expect(screen.getByText(/problem/i)).toBeVisible();
```

Reply:
i dont think this is what we are looking for? we want to make sure this is an alumni specific error message

Reply:
typo, the point I'm trying to make is to use getByText and not queryByText. queryByText should only be used when checking for non-existence.

```suggestion
  expect(screen.getByText(/alumni/i)).toBeVisible();
```